microcosm_flask/swagger/parameters/base.py

```diff
@@ -0,0 +1,142 @@
+from abc import ABC, abstractmethod
+from pkg_resources import iter_entry_points
+from typing import Any, Iterable, Mapping, Optional, Sequence, Tuple
+
+from marshmallow.fields import Field, List, Nested
+
+from microcosm_flask.naming import name_for
+from microcosm_flask.swagger.naming import type_name
+
+
+ENTRY_POINT = "microcosm_flask.swagger.parameters"
+
+
+class ParameterBuilder(ABC):
+    """
+    Plugin-aware swagger parameter builder.
+
+    Discovers builder subclasses via the `microcosm_flask.swagger.parameters` entry point
+    and delegates to the first compatible implementation.
+
+    """
+    def __init__(self):
+        self.parsers = {
+            "default": self.parse_default,
+            "description": self.parse_description,
+            "enum": self.parse_enum_values,
+            "format": self.parse_format,
+            "$ref": self.parse_ref,
+            "type": self.parse_type,
+            "items": self.parse_items,
+        }
+
+    @classmethod
+    def build(cls, field: Field) -> Mapping:
+        """
+        Build a swagger parameter from a marshmallow field.
+
+        """
+        builders = [
+            entry_point.load()()
```

Comment:
We can customize how we generate fields by registering a new entry point (e.g. in some other code base)